app/controllers/api_controller/arbitration_profiles.rb

```diff
@@ -3,39 +3,27 @@ module ArbitrationProfiles
     def create_resource_arbitration_profiles(_type, _id, data)
       validate_profile_data(data)
       attributes = data.dup
-      fetch_set_availability_zone(attributes)
-      fetch_set_provider(attributes)
+      attributes['ext_management_system'] = fetch_provider(provider_from_data(data)) if provider_from_data(data)
+      attributes['availability_zone'] = fetch_availability_zone(data['availability_zone']) if data['availability_zone']
+      attributes.delete('provider')
       arbitration_profile = collection_class(:arbitration_profiles).create(attributes)
-      if arbitration_profile.invalid?
-        raise BadRequestError, "Failed to add new arbitration profile -
-            #{arbitration_profile.errors.full_messages.join(', ')}"
-      end
+      validate_profile(arbitration_profile)
```

Comment:
@imtayadeway let me know how this method looks now. Moved the validation to another method because rubocop was mad at me about complexity.